target_link_options_if_exists.cmake

```diff
@@ -0,0 +1,45 @@
+# This file is part of Desktop App Toolkit,
+# a set of libraries for developing nice desktop applications.
+#
+# For license and copyright information please follow this link:
+# https://github.com/desktop-app/legal/blob/master/LEGAL
+
+include(CheckCXXCompilerFlag)
+
+function(target_link_options_if_exists target_name)
+    set(list ${ARGV})
+    list(REMOVE_AT list 0)
```

Comment:
It may be interesting. You can refer to ${ARGN} directly not altering the complete arguments list.
https://cmake.org/cmake/help/latest/command/function.html#arguments